tests/components/aurora_abb_powerone/test_config_flow.py

```diff
@@ -150,16 +162,111 @@ async def test_form_invalid_com_ports(hass):
+    TEST_DATA = {"device": "/dev/ttyUSB7", "address": 3, "name": "MyAuroraPV"}
+
+    with patch("aurorapy.client.AuroraSerialClient.connect", return_value=None,), patch(
+        "aurorapy.client.AuroraSerialClient.serial_number",
+        return_value="9876543",
+    ), patch(
+        "aurorapy.client.AuroraSerialClient.version",
+        return_value="9.8.7.6",
+    ), patch(
+        "aurorapy.client.AuroraSerialClient.pn",
+        return_value="A.B.C",
+    ), patch(
+        "aurorapy.client.AuroraSerialClient.firmware",
+        return_value="1.234",
+    ), patch(
+        "homeassistant.components.aurora_abb_powerone.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": config_entries.SOURCE_IMPORT}, data=TEST_DATA
+        )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+    assert result["data"][CONF_PORT] == "/dev/ttyUSB7"
+    assert result["data"][CONF_ADDRESS] == 3
+
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_import_night(hass):
+    """Test .yaml import when the inverter is inaccessible (e.g. darkness)."""
+    TEST_DATA = {"device": "/dev/ttyUSB7", "address": 3, "name": "MyAuroraPV"}
+
+    # First time round, no response.
     with patch(
-        "homeassistant.components.generic.camera.GenericCamera.async_camera_image",
-        return_value=None,
-    ):
+        "aurorapy.client.AuroraSerialClient.connect",
+        side_effect=AuroraError("No response after"),
+    ) as mock_setup_entry:  # , pytest.raises(ConfigEntryNotReady):
         result = await hass.config_entries.flow.async_init(
-            DOMAIN, context={"source": config_entries.SOURCE_IMPORT}, data=TESTDATA
+            DOMAIN, context={"source": config_entries.SOURCE_IMPORT}, data=TEST_DATA
         )
+
+    configs = hass.config_entries.async_entries(DOMAIN)
+    assert len(configs) == 1
+    entry = configs[0]
+    assert not entry.unique_id
+    assert entry.state == ConfigEntryState.SETUP_RETRY
+
+    assert len(mock_setup_entry.mock_calls) == 1
     assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
     assert result["data"][CONF_PORT] == "/dev/ttyUSB7"
     assert result["data"][CONF_ADDRESS] == 3
+
+    # Second time round, talking this time.
+    with patch("aurorapy.client.AuroraSerialClient.connect", return_value=None,), patch(
+        "aurorapy.client.AuroraSerialClient.serial_number",
+        return_value="9876543",
+    ), patch(
+        "aurorapy.client.AuroraSerialClient.version",
+        return_value="9.8.7.6",
+    ), patch(
+        "aurorapy.client.AuroraSerialClient.pn",
+        return_value="A.B.C",
+    ), patch(
+        "aurorapy.client.AuroraSerialClient.firmware",
+        return_value="1.234",
+    ), patch(
+        "homeassistant.components.aurora_abb_powerone.config_flow._LOGGER.getEffectiveLevel",
```

Comment:
We can use the `caplog` fixture to assert logs. But we don't seem to check the log.

Reply:
Agreed. Removed in #58887.